app/src/org/commcare/android/nsd/NSDDiscoveryTools.java

```diff
@@ -37,7 +41,8 @@
     public enum NsdState {
         Init,
         Discovery,
-        Idle
+        Idle,
+        Unsupported
```

Comment:
We actually won't know when we add the listener, which seems fine. The caller is asking to subscribe to events that will never happen, but I don't think that should cause any lifecycle issues.